### PR TITLE
clean up several js bits

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

* JavaScript code cleanup
* Potential formatting bugfix for AQL query explain output (for https://github.com/arangodb/release-3.6/issues/30#issuecomment-567443630)

- [x] Bug-Fix for *devel-branch* (i.e. no need for backports?)
- [ ] Bug-Fix for a *released version* (did you remember to port this to all relevant release branches?)
- [x] The behavior in this PR can be (and was) *manually tested* (support / qa / customers can test it)
- [ ] The behavior change can be verified via automatic tests

### Testing & Verification

This change is a trivial rework / code cleanup without any test coverage.

https://172.16.10.101/view/PR/job/arangodb-matrix-pr/7782/

### CODE REVIEW

File: js/client/modules/@arangodb/arango-database.js [hunk 1]
```diff
@@ -1232,7 +1216,7 @@ ArangoDatabase.prototype._createView = function (name, type, properties) {
   var nname = requestResult.name;
 
   if (nname !== undefined) {
-    this._registerView(nname, new this._viewConstructor(this, requestResult));
+    this[nname] = new ArangoView(this, requestResult);
```
Comment (hkernbach) [inline js/client/modules/@arangodb/arango-database.js]: I know that this is not your fault, but what is `nname`. Not clear without reading the whole code. Maybe we could change that.

File: js/common/modules/@arangodb/aql/explainer.js [hunk 2]
```diff
@@ -766,15 +766,15 @@ function processQuery(query, explain, planIndex) {
...
 
-        if (String(n.calls).length > maxCallsLen) {
-          maxCallsLen = String(n.calls).length;
+        if (String(nodes[n.id].calls).length > maxCallsLen) {
+          maxCallsLen = String(nodes[n.id].calls).length;
         }
-        if (String(n.items).length > maxItemsLen) {
-          maxItemsLen = String(n.items).length;
+        if (String(nodes[n.id].items).length > maxItemsLen) {
+          maxItemsLen = String(nodes[n.id].items || 0).length;
```
Comment (hkernbach) [inline js/common/modules/@arangodb/aql/explainer.js]: duplicate check, see Row: 771

Comment (hkernbach) [inline js/common/modules/@arangodb/aql/explainer.js]:
```suggestion
          maxItemsLen = String(nodes[n.id].items).length;
```

**Comment by jsteemann** [comment]
https://172.16.10.101/view/PR/job/arangodb-matrix-pr/7784/

**Comment by jsteemann** [comment]
https://172.16.10.101/view/PR/job/arangodb-matrix-pr/7785/

File: js/client/modules/@arangodb/arango-database.js (hunk 1 above)
Comment (jsteemann) [reply js/client/modules/@arangodb/arango-database.js]: Ok, fixing it...

File: js/common/modules/@arangodb/aql/explainer.js (hunk 2 above)
Comment (jsteemann) [reply js/common/modules/@arangodb/aql/explainer.js]: Fixed now.

**Pull request merged**